modules/apps/headless/headless-asset-library/headless-asset-library-test/src/testIntegration/java/com/liferay/headless/asset/library/resource/v1_0/test/AssetLibraryResourceTest.java

```diff
@@ -82,6 +86,122 @@ public void testPostAssetLibraryLinkToSite() throws Exception {
 		_assertLinkedSites(assetLibrary);
 	}
 
+	@Override
+	protected void assertValid(AssetLibrary assetLibrary) throws Exception {
+		boolean valid = true;
+
+		if ((assetLibrary.getDateCreated() == null) ||
+			(assetLibrary.getDateModified() == null) ||
+			(assetLibrary.getId() == null)) {
+
+			valid = false;
+		}
+
+		DepotEntry depotEntry = _depotEntryLocalService.getDepotEntry(
+			assetLibrary.getId());
+
+		Group group = depotEntry.getGroup();
```

Comment:
Why do we need this? Shouldn't testDepotEntry.getDepotEntryId() be the same as assetLibrary.getId() (i.e. the represent the same thing under the hood).

Reply:
No, we just created a new asset library on the post, it's not the automatically created by the test.

The new asset Library is not the same as testDepotEntry, and the new asset library has no relation with the testGroup.

This is a tricky thing, because the autogenerated code in not ready to deal with new asset libraries.

Reply:
changing the commit message to be clearer <3 thanks!